samples/FallbackSendFile/SendFileMiddleware.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace FallbackSendFile
+{
+    /// <summary>
+    /// Middleware which adds an <see cref="IHttpSendFileFeature"/> implementation if currently missing.
+    /// </summary>
+    public class SendFileMiddleware
```

Comment:
Do not plan to add tests of this sample. That would require odd test dependencies e.g. on `TestHost`. And Kestrel, TestHost and WebListener all support `IHttpSendFileFeature` without any help.